### PR TITLE
feat: add license_type

### LINKED ISSUE #139: Add license_type
### Module Name

vm

### Description

Add license_type that is needed by Windows VMs

### Example Terraform Configuration

```terraform
variable "license_type" {
  description = "The license type to assign to this VM."
  type        = string
  default     = null
}

...
license_type        = var.license_type
...
```

### PR DESCRIPTION
**Pull request by stiross79**
Fixes: equinor/terraform-baseline#139

### CODE REVIEW

**Comment by hknutsen** [comment]
I see that Linux VMs support this attribute as well: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine#license_type

Should implement that at the same time?

**Comment by stiross79** [comment]
> I see that Linux VMs support this attribute as well: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine#license_type
> 
> Should implement that at the same time?

You are correct. We have never used it for Linux VMs 😄 . I will add it.

**Review by sebastianlolv — DISMISSED** [review]
LGTM 👍

**Comment by stiross79** [comment]
> LGTM 👍

Unfortunately the test fails. The `license_type` variable validation block does not like the default `null` value.

File: variables.tf [hunk 1]
```diff
@@ -221,3 +221,14 @@ variable "availability_set_id" {
...
 }
+
+variable "license_type" {
+  description = "Specifies the type of license which should be used for this Virtual Machine. Value must be \"None\", \"Windows_Client\", \"Windows_Server\", \"RHEL_BYOS\" or \"SLES_BYOS\". Defaults to null."
+  type        = string
+  default     = null
+
+  validation {
+    condition     = contains(["None", "Windows_Client", "Windows_Server", "RHEL_BYOS", "SLES_BYOS"], var.license_type)
+    error_message = "License mode must be \"None\", \"Windows_Client\", \"Windows_Server\", \"RHEL_BYOS\" or \"SLES_BYOS\"."
```
Comment (hknutsen) [inline variables.tf]: @stiross79 Try something like this:

```suggestion
    condition     = contains(["None", "Windows_Client", "Windows_Server", "RHEL_BYOS", "SLES_BYOS", null], var.license_type)
    error_message = "License mode must be \"None\", \"Windows_Client\", \"Windows_Server\", \"RHEL_BYOS\", \"SLES_BYOS\" or null."
```

You'll have to do something similar for the resource preconditions.

**Comment by stiross79** [comment]
@hknutsen Works like a charm 😄

**Review by stiross79 — COMMENTED** [review]
Looks Good 🎉

**Review by sebastianlolv — APPROVED** [review]
Looks Even better this time 😅

**Pull request merged**